capa/features/extractors/base_extractor.py

```diff
@@ -296,6 +298,19 @@ def extract_insn_features(
         raise NotImplementedError()
 
 
+def FunctionFilter(extractor: StaticFeatureExtractor, functions: Set) -> StaticFeatureExtractor:
+    get_functions = extractor.get_functions  # fetch original get_functions()
+
+    def filtered_get_functions(self):
+        yield from (f for f in get_functions() if f.address in functions)
+
+    # make a copy of the extractor before decorating the get_functions() method
+    new_extractor = copy(extractor)
```

Comment:
Rather than a comment that describes *what* is done, please add a comment that describes why you make this copy. I don't quite understand it yet, so this context would be useful to any future reader.

Reply:
and same comment below.

Reply:
I've added a comment now. Please let me know if it's not clear enough.